### PR TITLE
sssmokin

### PR DESCRIPTION
**Pull request by jupyterkat**
## About The Pull Request

some really simple implementation of smoke, makes you cough, blocks your vision, only appears on things burning.
can be get rid of through scrubbers like any sane gas

## Why It's Good For The Game

it's hip to smoke

## Changelog
:cl:
add: added actual smoke
/:cl:

### CODE REVIEW

File: austation/code/controllers/subsystem/fire_burning.dm [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+		src.currentrun = processing.Copy()
+
+	//cache for sanic speed (lists are references anyways)
+	var/list/currentrun = src.currentrun
+	var/delta_time = wait * 0.1
+
+	while(currentrun.len)
+		var/obj/O = currentrun[currentrun.len]
+		currentrun.len--
+		if (!O || QDELETED(O))
```
Comment (TheFakeElon) [inline austation/code/controllers/subsystem/fire_burning.dm:12]: QDELETED checks that
```suggestion
		if (QDELETED(O))
```

File: tgui/public/tgui.bundle.css [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
...wyLjI3ODE1IDMuMzQwODEsMy4xMDYwMiA0LjgzOTM5LDYuNzc0OTEgOC44NDkzNCwxNi4yNDU2NiAxMi4wMjk1MSwyOC40MTM5NyBsIDIwLjUzMjM0LDAgMCwtNTUuOTk5NjcgYyAtNi42NzczMSwtNC41OTM4MSAtMTkuODM2NDMsLTEwLjQ3MzA5IC0zNi4xNDA3MSwtMTUuODI1MjIgeiBtIC0yOC4xMjA0OSw1LjYwNTUxIDguNTY0NzksMTcuNzE2NTUgYyAtMTEuOTcwMzcsLTYuNDY2OTcgLTEzLjg0Njc4LC05LjcxNzI2IC04LjU2NDc5LC0xNy43MTY1NSB6IG0gMjIuNzk3MDUsMCBjIDIuNzcxNSw3Ljk5OTI5IDEuNzg3NDEsMTEuMjQ5NTggLTQuNDkzNTQsMTcuNzE2NTUgbCA0LjQ5MzU0LC0xNy43MTY1NSB6IG0gMTUuMjIxOTUsMjQuMDA4NDggOC41NjQ3OSwxNy43MTY1NSBjIC0xMS45NzAzOCwtNi40NjY5NyAtMTMuODQ2NzksLTkuNzE3MjYgLTguNTY0NzksLTE3LjcxNjU1IHogbSAyMi43OTcwNCwwIGMgMi43NzE1LDcuOTk5MjkgMS43ODc0MSwxMS4yNDk1OCAtNC40OTM1NCwxNy43MTY1NSBsIDQuNDkzNTQsLTE3LjcxNjU1IHogbSAtOTkuMTEzODQsMi4yMDc2NCA4LjU2NDc5LDE3LjcxNjU1IGMgLTExLjk3MDM4MiwtNi40NjY5NyAtMTMuODQ2NzgyLC05LjcxNzI2IC04LjU2NDc5LC0xNy43MTY1NSB6IG0gMjIuNzk1NDIsMCBjIDIuNzcxNSw3Ljk5OTI5IDEuNzg3NDEsMTEuMjQ5NTggLTQuNDkzNTQsMTcuNzE2NTUgbCA0LjQ5MzU0LC0xNy43MTY1NSB6IiAvPg0KPC9zdmc+DQo8IS0tIFRoaXMgd29yayBpcyBsaWNlbnNlZCB1bmRlciBhIENyZWF0aXZlIENvbW1vbnMgQXR0cmlidXRpb24tU2hhcmVBbGlrZSA0LjAgSW50ZXJuYXRpb25hbCBMaWNlbnNlLiAtLT4NCjwhLS0gaHR0cDovL2NyZWF0aXZlY29tbW9ucy5vcmcvbGljZW5zZXMvYnktc2EvNC4wLyAtLT4NCg==)}
```
Comment (TheFakeElon) [inline tgui/public/tgui.bundle.css:12]: What are these for?

File: code/modules/atmospherics/gasmixtures/auxgm.dm [hunk 3]
```diff
@@ -71,7 +71,7 @@ GLOBAL_LIST_INIT(nonreactive_gases, typecacheof(list(GAS_O2, GAS_N2, GAS_CO2, GA
 			visibility[g] = gas.moles_visible
 			overlays[g] = new /list(FACTOR_GAS_VISIBLE_MAX)
 			for(var/i in 1 to FACTOR_GAS_VISIBLE_MAX)
-				overlays[g][i] = new /obj/effect/overlay/gas(gas.gas_overlay, i * 255 / FACTOR_GAS_VISIBLE_MAX)
+				overlays[g][i] = new /obj/effect/overlay/gas(gas.gas_overlay, i * 255 / FACTOR_GAS_VISIBLE_MAX, gas.custom_icon, gas.custom_x, gas.custom_y, gas.vis_flags, gas.layers) //austation -- added 5 new args
```
Comment (TheFakeElon) [inline code/modules/atmospherics/gasmixtures/auxgm.dm:74]: Would appreciate a comment stating as to why, or at least in what PR :)

File: code/modules/atmospherics/gasmixtures/auxgm.dm [hunk 4]
```diff
@@ -124,7 +124,19 @@ GLOBAL_DATUM_INIT(gas_data, /datum/auxgm, new)
 	appearance_flags = TILE_BOUND
 	vis_flags = NONE
 
-/obj/effect/overlay/gas/New(state, alph)
+//austation begin -- added 3 new args and vars
+/obj/effect/overlay/gas/New(state, alph, custom_icon, _pixel_x, _pixel_y, _flags, _layer)
```
Comment (TheFakeElon) [inline code/modules/atmospherics/gasmixtures/auxgm.dm:128]: same here

File: austation/code/modules/atmospherics/gasmixtures/auxgm.dm [hunk 5]
```diff
@@ -0,0 +1,6 @@
+/datum/gas
+	var/custom_icon = null //custom icon filepath for snowflakey gases
+	var/custom_x = null
+	var/custom_y = null
+	var/vis_flags = null
+	var/layers = null
```
Comment (TheFakeElon) [inline austation/code/modules/atmospherics/gasmixtures/auxgm.dm:6]: Wouldn't it be better to make this it's own subtype? Adding all of these (moreso all of checks they add in new) adds unneeded overhead to gas creation

File: code/modules/atmospherics/gasmixtures/auxgm.dm [hunk 6]
```diff
@@ -124,7 +124,19 @@ GLOBAL_DATUM_INIT(gas_data, /datum/auxgm, new)
...
+	if(custom_icon)
+		icon = custom_icon
+	if(_pixel_x)
+		pixel_x = _pixel_x
+	if(_pixel_y)
+		pixel_y = _pixel_y
+	if(_flags)
+		appearance_flags = _flags
+	if(_layer)
+		layer = _layer
```
Comment (TheFakeElon) [inline code/modules/atmospherics/gasmixtures/auxgm.dm:141]: They'll be null anyway, may as well save the performance
```suggestion
	icon = custom_icon
	pixel_x = _pixel_x
	pixel_y = _pixel_y
	appearance_flags = _flags
	layer = _layer
```

Comment (jupyterkat) [reply code/modules/atmospherics/gasmixtures/auxgm.dm:141]: no, we don't want them to be null if it's null, then it'll make it null for every gas that we want the default thing for

File: austation/code/modules/atmospherics/gasmixtures/auxgm.dm (hunk 5 above)
Comment (jupyterkat) [reply austation/code/modules/atmospherics/gasmixtures/auxgm.dm:6]: there's already a shit ton of vars, a few more won't do shit

File: code/modules/atmospherics/gasmixtures/auxgm.dm (hunk 3 above)
Comment (jupyterkat) [reply code/modules/atmospherics/gasmixtures/auxgm.dm:74]: for the gases to have specific layers and have specific x and y offsets, the smoke sprite is 96x96

**Comment by MCterra10** [comment]
DNM until reviews are actually handled instead of resolved and the code left unchanged.

**Pull request closed without merging**